### ISSUE TITLE
[beta] some "get a hint" links still broken

### ISSUE THREAD

**Issue by moT01** [freeCodeCamp/freeCodeCamp#35155]
Some of the "get a hint" links are still not working in beta. It seems to be ones that had possibly been previously fixed by renaming the guide folder.

Here's a list of challenges with broken hint links...
**node:**
https://www.freecodecamp.rocks/learn/apis-and-microservices/managing-packages-with-npm/how-to-use-package.json-the-core-of-any-node.js-project-or-npm-package
**D3:**
https://www.freecodecamp.rocks/guide/certifications/data-visualization/data-visualization-with-d3/use-the-d3.max-and-d3.min-functions-to-find-minimum-and-maximum-values-in-a-dataset
**All react lessons with a period in the title**

Basically it looks like the guide folders have been renamed at some point to replace the periods with a dash, and I think a fix is to revert the folder title's to have the periods again. 

**PR created:**
- [x] arabic
- [x] chinese
- [x] english
- [x] portuguese
- [x] russian
- [x] spanish

**PR merged:**
- [x] arabic
- [x] chinese
- [x] english
- [x] portuguese
- [x] russian
- [x] spanish

**Comment by RandellDawson** [comment]
@moT01 There is no reason to rename those files (the .  They should work, but you can not use Windows to test it locally, because it sees the dots and tries to serve as file.  @ValeraS fixed this issue in another PR.   I will let @ValeraS shed more light on this issue.

**Comment by moT01** [comment]
Quincy mentioned that as a fix on one of those issues, another one has an open PR with that as a fix, and [this one](https://github.com/freeCodeCamp/freeCodeCamp/issues/18010) merged a fix like that. That's why I thought to go that route. So, I suppose, you just redirect to the current URL instead of changing the names. Yea, that's probably a lot easier. It might make some of the links where the folder name was already changed not work.

**Comment by RandellDawson** [comment]
@moT01 So I noticed you updated the body of the issue, so my first comment is irrelevant now.  If I am ready this issue correctly now, we just need to rename those two guide articles to include the dots `.` again and everything will be back to normal for those challenges.  If so, have you created a PR yet.  If not, I don't mind creating it.

**Comment by moT01** [comment]
Yea, sorry about that - I was going to make a note about that. It's all the react lessons with periods, as well. I was just testing locally to make sure it all works, and was going to make a PR. Do you think I should make changes to the folders in all languages? I would suspect the same issue occurs there.

**Comment by RandellDawson** [comment]
I imagine so, but put each language changes on a separate PR if you don't mind.

**Comment by RandellDawson** [comment]
Just to clarify, you are only going to be renaming guide filenames?

**Comment by moT01** [comment]
Yes, that's the plan if I can test it locally to make it sure it's working. 
So...
https://github.com/freeCodeCamp/freeCodeCamp/tree/master/guide/english/certifications/apis-and-microservices/managing-packages-with-npm/how-to-use-package-json-the-core-of-any-node-js-project-or-npm-package 

would become
https://github.com/freeCodeCamp/freeCodeCamp/tree/master/guide/english/certifications/apis-and-microservices/managing-packages-with-npm/how-to-use-package.json-the-core-of-any-node.js-project-or-npm-package

**Comment by RandellDawson** [comment]
You can not use Windows to test the Hint button will properly link to any guide article which has a dot in the folder name.  I discovered this last week.  You need to test on Linux or maybe Mac.

**Comment by RandellDawson** [comment]
Also, make sure to copy `guide/english/certifications` folder over to `mock-guide/english`, so your local build will have all the guide articles.

**Comment by RandellDawson** [comment]
All PRs involving the fixes to this issue have now been merged, so closing this PR.
